.github/workflows/spack.yml

```diff
@@ -0,0 +1,38 @@
+name: spack
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+  pull_request_review:
+    types: 
+      - submitted
+  workflow_dispatch:
+
+jobs:
+  gcc10-openmpi416:
+    runs-on: [self-hosted, triloamd02]
```

Comment:
@sebrowne @fryeguy52 @jmlapre 

Before we get this merged in, we should settle on a labeling scheme for our AT2 runners so that they can be picked up correctly in the future ([see runners here](https://github.com/trilinos/Trilinos/settings/actions/runners)).

My current plan is to just try to follow a pattern similar to our current Jenkins jobs, so for the current triloamd01 spawner with the gcc 10.3.0 openmpi 4.1.6 container, the labels applied would be `gcc-10.3.0_openmpi-4.1.6` along with a label of the hostname. 

The other option would be to split that label into its components with `gcc-10.3.0`, `openmpi-4.1.6` and have workflow explicitly find a runner that matches all the labels. This method would allow flexibility in reusing these runners for serial jobs too where the serial workflow file would only specify `gcc-10.3.0` and use a serial config.

What do ya'll think?

Reply:
I like using the "toolchain" string for now.  Meaning `gcc-10.3.0_openmpi-4.1.6`, `cuda-11.2.0_gcc-10.3.0_openmpi-4.1.6`, `gcc-10.3.0` would be what we're using soon.  Then the job can use runners on either machine as long as it's the correct container.

Reply:
Spoke with @achauphan , we're going forwards with the previous comment approach.